electrode_models/nucleation_single_particle.py

```diff
@@ -0,0 +1,291 @@
+        # electode (zero for anode, n_vars_anode + n_vars_sep for the cathode)
+        self.SV_offset = offset
+
+        # Determine the electrode capacity (Ah/m2)
+
+        # Max concentration of stored ion (intercalated Li)
+        # Save initial X
+        #X_o = self.bulk_obj.X 
+        # Set object concentration to fully lithiated:
+        #self.bulk_obj.X = inputs['stored-ion']['name']+':1.0' 
+        # Concentration of stored Li, per unit volume of intercalation phase:
+        #Conc = self.bulk_obj[inputs['stored-ion']['name']].concentrations[0]
+        
+        self.capacity = (inputs['stored-species']['charge']*ct.faraday
+                 *self.eps_elyte)*inputs['thickness']/(3600
+                 *inputs['stored-species']['MW']) 
+
+        # Return Cantera object composition to original value:
+        #self.bulk_obj.X = X_o
+        
+        # Number of state variables: electrode potential, electrolyte, porosity 
+        # potential, electrode composition (n_species), electrolyte composition 
+        # (n_species)
+        self.n_vars = 3 + self.elyte_obj.n_species
+
+        # This model produces one plot, for the intercalation concentration.
+        self.n_plots = 0
+
+        # Set Cantera object state:
+        self.bulk_obj.electric_potential = inputs['phi_0']
+        # If the user provided an initial composition, use that, here:
+        if 'X_0' in inputs:
+            self.bulk_obj.TPX = (params['T'], params['P'], inputs['X_0'])
+        else:
+            self.bulk_obj.TP = params['T'], params['P']
+
+        self.elyte_obj.TP = params['T'], params['P']
+        self.surf_obj.TP = params['T'], params['P']
+        #self.conductor_obj.TP = params['T'], params['P']
+
+        # Set up pointers to specific variables in the solution vector:
+        self.SVptr = {}
+        self.SVptr['phi_ed'] = np.array([0])
+        self.SVptr['phi_dl'] = np.array([1])
+        self.SVptr['eps_oxide'] = np.array([2])
+        self.SVptr['C_k_elyte'] = np.arange(3, 3 + self.elyte_obj.n_species)
+        
+        # There is only one node, but give the pointer a shape so that SVptr
+        # ['C_k_elyte'][j] accesses the pointer array:
+        self.SVptr['C_k_elyte'].shape = (1,self.elyte_obj.n_species)
+
+        # A pointer to where the SV variables for this electrode are, within the 
+        # overall solution vector for the entire problem:
+        self.SVptr['electrode'] = np.arange(offset, offset+self.n_vars)
+
+        # Save the indices of any algebraic variables:
+        self.algvars = offset + self.SVptr['phi_ed'][:]
+
+    def initialize(self, inputs, sep_inputs):
+
+        # Initialize the solution vector for the electrode domain:
+        SV = np.zeros([self.n_vars])
+
+        # Load intial state variables:
+        SV[self.SVptr['phi_ed']] = inputs['phi_0']
+        SV[self.SVptr['phi_dl']] = sep_inputs['phi_0'] - inputs['phi_0']
+        SV[self.SVptr['eps_oxide']] = self.eps_oxide 
```

Comment:
Rather than storing this initial `eps_oxide` in self and using it just once, maybe just let the user specify it as part of the input file, then just load `inputs['eps_oxide_init']` (or somesuch) on this line.